### PR TITLE
Actually get dummy test app running properly and passing feature tests

### PR DESCRIPTION
**Pull request by jrochkind**
* Configure dummy test all with the test controller/views for demo under test
* properly set up sprockets for bootstrap 4 to include JS in dummy app
* properly get all dummy app dependencies set up -- this is the weirdest most confusing part
  * dependencies are actually listed in .gemspec as development dependencies
  * but may also need to be require'd in application.rb, which would normally happen automatically if they were
    in a real Gemfile

* We give up on being able to test both bootstrap 3 and 4 -- that's not really tenable with the dummy app approach, since they use different gems entirely, and different local CSS files. We only test bootstrap 4 now.

* Actually enable the formerly disabled feature tests!

-------

Includes in this PR one commit:

FileSystem driver has new `allow_relative_home` param

If set to true, then `home` can be a relative path (relative to Rails.root).

The use case is mainly for CI/dummy test app setup. Added new param to make it opt-in, to make sure it's completely backwards compat and does not introduce any potential security issues -- you need to opt in to new supported behavior with new allow_relative_home param.

### CODE REVIEW

**Comment by jrochkind** [comment]
Ok getting very close!

But we were trying to configure browse_everything_providers.yml, under test, with a `file_system` provider pointing to test app source. 

But it can only take an absolute path?  And we need to commit to repo... where the absolute path is going to be different depending on where the repo is checked out, etc. 

Hmm. We may expand it to take a relative path (to CWD)... if opted into?

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 version: 2.1
 orbs:
   samvera: samvera/circleci-orb@1.0
+  browser-tools: circleci/browser-tools@1.1
```
Comment (jrgriffiniii) [inline .circleci/config.yml:5]: 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -19,6 +20,8 @@ jobs:
     environment:
       RAILS_VERSION: << parameters.rails_version >>
     steps:
+      - browser-tools/install-chrome
```
Comment (jrgriffiniii) [inline .circleci/config.yml:23]: 👍

File: spec/dummy_test_app/config/browse_everything_providers.yml [hunk 3]
```diff
@@ -3,7 +3,8 @@
 # The file_system provider can be a path to any directory on the server where your application is running.
 #
 file_system:
-  home: /Users/jrochkind/code/browse-everything/.internal_test_app
```
Comment (jrgriffiniii) [inline spec/dummy_test_app/config/browse_everything_providers.yml:6]: I am sorry for missing this line within a previous pull request review.

File: spec/features/select_files_spec.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
   shared_examples 'browseable files' do
     # This is a work-around until the support for Webpacker is resolved
-    xit 'selects files from the filesystem' do
+    it 'selects files from the filesystem' do
```
Comment (jrgriffiniii) [inline spec/features/select_files_spec.rb:10]: 👍

File: spec/features/test_compiling_stylesheets_spec.rb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 describe 'Compiling the stylesheets', type: :feature, js: true do
-  xit 'does not raise errors' do
+  it 'does not raise errors' do
```
Comment (jrgriffiniii) [inline spec/features/test_compiling_stylesheets_spec.rb:4]: 👍

**Review by jrgriffiniii — APPROVED** [review]
This is phenomenal, thank you very much again for undertaking this much needed improvement!

**Pull request merged**